Add log correlation documentation (#481)
signalfx/splunk-otel-go: README.md

otel.SetTextMapPropagator(propagation.TraceContext{})
```

## Log Correlation

## Correlate Traces and Logs

> Trace metadata can be added to logs using the OpenTelemetry trace API. Added
> trace metadata to log events allows logs to be explored from the Splunk
> Observability Cloud.

Reviewer: ```suggestion
You can add trace metadata to logs using the OpenTelemetry trace API. Trace metadata lets 
you explore logs in Splunk Observability Cloud.
```
